### DISCUSSION TITLE
Make RenameClassRector more flexible by supporting callbacks?

### DISCUSSION THREAD

**Discussion by dorrogeray** [rectorphp/rector#7556] [Ideas]
Hi, I would like to write a rector rule which renames classes extending the `Exception` class to have suffix `Exception`, and possibly similar thing for interfaces. I was thinking whether or not it would be a good idea to add support of callbacks to the `RenameClassRector` instead of writing multiple independent rules, which serve similar purpose? Any thoughts on this?

**Comment by TomasVotruba** [comment]
Hi, how exactly would that look like?

**Reply by dorrogeray** [reply]
Hi, sorry for the delayed response. I have prepared this draft https://github.com/rectorphp/rector-src/pull/3023
